### PR TITLE
Event recorder

### PR DESCRIPTION
**Pull request by njuettner**
Add events in cluster-operator

It might be that it will be removed in the future or we have to transition those events in another operator but I think it would give us some value to see when things get created or updated.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by njuettner** [comment]
Tested in gauss, this add a bit more context.

image

**Comment by njuettner** [comment]
Tested in gauss 
image

File: service/controller/resource/statuscondition/create.go [hunk 1]
```diff
@@ -40,6 +41,23 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "found latest cluster")
 	}
 
+	var cl apiv1alpha2.Cluster
```
Comment (njuettner) [inline service/controller/resource/statuscondition/create.go:44]: I included upstream cluster to write events on clusters.cluster.x-k8s.io CR

**Review by anvddriesch — APPROVED** [review]
LGTM

**Pull request merged**